BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -1748,6 +1764,13 @@ IEnumerator BoostRoutine()
                     }
 
                 //thrust
+                if (BDArmorySettings.MISSILE_FUEL_DEDUCTION && burnRate > 0 && burnedMass < boosterMass)
+                {
+                    burnedMass = Mathf.Min(burnedMass + burnRate, boosterMass);
+                    ordinanceMass = -burnedMass;
+                    part.UpdateMass();
+                    part.Update();
+                }
                 if (spoolEngine)
```

Comment:
This part is unnecessary; it'd need to only do ``ordinanceMass -= BurnRate;``, as burnRate already supplies the mass flow value per tick, and IPartMassModifier will automatically update part mass as the ordinanceMass value changes